core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -299,39 +303,48 @@ public void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploym
+                Optional<ConfigChangeSetExecutor> configChangeSetExecutor =
+                        ConfigChangeSetFactory.getExecuter(osCategory);
+                if (configChangeSetExecutor.isPresent()) {
+                    // Initialize config set repos on agent
+                    if (configChangeSetExecutor.get().initConfigChangeSet(testPlan)) {
+                        for (ConfigChangeSet configChangeSet : configChangeSetList) {
+                            logger.info("Start running config change set for " + configChangeSet.getName());
+                            // Apply config change set script on agent
+                            if (configChangeSetExecutor.get().applyConfigChangeSet(testPlan, configChangeSet,
+                                            deploymentCreationResult, true)) {
+                                // Run test scenarios for relevant config change set
+                                for (TestScenario testScenario : testPlan.getTestScenarios()) {
+                                    if (configChangeSet.getName().equals(testScenario.getConfigChangeSetName())) {
+                                        executeTestScenario(testScenario, deploymentCreationResult, testPlan);
+                                    }
+                                }
+                                // Revert config change set after running test scenarios
+                                // If revert back is not success then, break all test scenarios running
+                                if (!configChangeSetExecutor.get().applyConfigChangeSet(testPlan, configChangeSet,
+                                        deploymentCreationResult, false)) {
+                                    logger.info("Config change set revert script execution failed for " +
+                                            configChangeSet.getName() + ". Abort running test scenarios");
+                                    break;
+                                }
+                            }
                         }
-                    } else {
-                        testScenario.setStatus(Status.ERROR);
-                        logger.error(String.format("Error parsing the results for the " +
-                                "scenario %s", testScenario.getName()));
+                        // Remove config set repos on agent
+                        configChangeSetExecutor.get().deInitConfigChangeSet(testPlan);
                     }
-                } catch (ScenarioExecutorException e) {
-                    // we need to continue the rest of the tests irrespective of the exception thrown here.
-                    logger.error(StringUtil.concatStrings("Error occurred while executing the SolutionPattern '",
-                            testScenario.getName(), "' in TestPlan\nCaused by "), e);
+                } else {
+                    logger.error("Unable to find a Tinker Executor for OS category " + osCategory);
```

Comment:
Shall we add more details about the situation such as testplan-id, infra-combination, etc?

Reply:
Fixed